lib/util/index.js

```diff
@@ -603,18 +630,26 @@ function createDimensionsPayload(dimensions) {
  * @returns {object}
  */
 function createRenderDimensionsPayload(renderDimensions) {
-  return [
-    ['dominantSpeaker', 'active_speaker']
-  ].concat(Object.values(constants.trackPriority).map(priority => {
-    return [priority];
-  })).reduce((payload, [prop, payloadProp = prop]) => {
-    const value = renderDimensions[prop];
-    if (typeof value === 'object') {
-      payload = Object.assign({
-        [payloadProp]: createDimensionsPayload(value)
-      }, payload);
-    }
-    return payload;
+  const { PRIORITY_HIGH, PRIORITY_LOW, PRIORITY_STANDARD } = constants.trackPriority;
+  return createRSPPayload(renderDimensions, [
+    { prop: 'dominantSpeaker', type: 'object', payloadProp: 'active_speaker', transform: createDimensionsPayload },
+    { prop: PRIORITY_HIGH, type: 'object', transform: createDimensionsPayload },
+    { prop: PRIORITY_LOW, type: 'object', transform: createDimensionsPayload },
+    { prop: PRIORITY_STANDARD, type: 'object', transform: createDimensionsPayload }
+  ]);
+}
+
+/**
+ * Create an RSP payload for the given object.
+ * @param {object} object
+ * @param {Array<object>} propConversions
+ * @returns {*}
+ */
+function createRSPPayload(object, propConversions) {
```

Comment:
nice refactor, but can use more beefed up description and documentation on parameters